keps/sig-node/2411-cri-container-log-rotation/README.md

```diff
@@ -0,0 +1,301 @@
+
+- **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+
+  Yes, but if disabled the container logs will not be rotated when using a CRI container runtime.
+
+- **What happens if we reenable the feature if it was previously rolled back?**
+
+  No impact, container log rotation will work for CRI container runtimes.
+
+- **Are there any tests for feature enablement/disablement?**
+
+  There are already unit and node e2e tests in place for this feature.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+- **How can a rollout fail? Can it impact already running workloads?**
+
+   When the container log manager doesn't start up and rotate the logs as expected. Restarts shouldn't affect ths as it is the container runtime that will be writing the logs. The kubelet is in charge of checking log size and rotating when needed.
+
+- **What specific metrics should inform a rollback?**
+
+  - There is major loss of logs on nodes that use a CRI runtime.
+  - Logging agents are unable to integrate with k8s and aggregate logs to a remote storage.
+
+- **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  
+  Any manual testing was done when the feature was initially implemented in https://github.com/kubernetes/kubernetes/pull/59898.
+
+- **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  
+  No
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+- **How can an operator determine if the feature is in use by workloads?**
+
+  When a CRI container runtime is used, the logs are being rotated and stored in the log directory structure with a gzip format.
+
+- **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  - [ ] Metrics
+    - Metric name:
+    - [Optional] Aggregation method:
+    - Components exposing the metric:
+  - [x] Other (treat as last resort)
+    - Details: TBD
+
+- **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  
+  100% of logs are rotated according the configured max size and files without any loss.
+
+- **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+
+  N/A
+
+### Dependencies
+
+_This section must be completed when targeting beta graduation to a release._
+
+- **Does this feature depend on any specific services running in the cluster?**
+
+  - [Kubelet]
+    - Usage description: Responsible for managing the log directory structure and rotating the logs.
+      - Impact of its outage on the feature: Logs will not be rotated. Container runtime may continue to write logs to the file even after the max size has been reached.
```

Comment:
ultimately can bring the server down if logs keep growing uncontrolled.

Reply:
unless eviction is enabled in which the kubelet would evict pods....

Reply:
Added a line mentioning this